samples/bluetooth/peripheral_nfc_pairing/src/main.c

```diff
@@ -66,96 +80,67 @@ static int tk_value_generate(void)
 {
-	int err;
-
-	/* Filter already connected peers. */
-	for (size_t i = 0; i < CONFIG_BT_MAX_CONN; i++) {
-		if (bt_conn_lookup_addr_le(BT_ID_DEFAULT, &info->addr)) {
-			return;
-		}
-	}
-
-	err = k_msgq_put(&bonds_queue, (void *) &info->addr, K_NO_WAIT);
-	if (err) {
-		printk("No space in the queue for the bond.\n");
-	}
-}
-
-static void advertising_start(void)
-{
-	k_msgq_purge(&bonds_queue);
-	bt_foreach_bond(BT_ID_DEFAULT, bond_find, NULL);
-
-	k_work_submit(&adv_work);
+	k_poll_signal_init(&pair_signal);
+	k_poll_event_init(&events[NFC_TNEP_EVENTS_NUMBER],
+			  K_POLL_TYPE_SIGNAL, K_POLL_MODE_NOTIFY_ONLY,
+			  &pair_signal);
 }
 
-/** .. include_startingpoint_pair_msg_rst */
-static int pairing_msg_generate(uint32_t *len)
+static int paring_key_generate(void)
 {
 	int err;
-	struct nfc_ndef_le_oob_rec_payload_desc rec_payload;
-	struct nfc_ndef_ch_msg_records ch_records;
-	uint32_t ndef_size = nfc_t4t_ndef_file_msg_size_get(*len);
 
-	NFC_NDEF_MSG_DEF(hs_msg, 2);
+	printk("Generating new pairing keys\n");
 
 	err = bt_le_oob_get_local(BT_ID_DEFAULT, &oob_local);
 	if (err) {
 		printk("Error while fetching local OOB data: %d\n", err);
 	}
 
-	err = tk_value_generate();
-	if (err) {
-		return err;
-	}
-
-	memset(&rec_payload, 0, sizeof(rec_payload));
-
-	rec_payload.addr = &oob_local.addr;
-	rec_payload.le_sc_data = &oob_local.le_sc_data;
-	rec_payload.tk_value = tk_value;
-	rec_payload.local_name = bt_get_name();
-	rec_payload.le_role = NFC_NDEF_LE_OOB_REC_LE_ROLE(
-		NFC_NDEF_LE_OOB_REC_LE_ROLE_PERIPH_ONLY);
-	rec_payload.appearance = NFC_NDEF_LE_OOB_REC_APPEARANCE(
-		CONFIG_BT_DEVICE_APPEARANCE);
-	rec_payload.flags = NFC_NDEF_LE_OOB_REC_FLAGS(BT_LE_AD_NO_BREDR);
+	return tk_value_generate();
+}
 
-	NFC_NDEF_LE_OOB_RECORD_DESC_DEF(oob_rec, '0', &rec_payload);
-	NFC_NDEF_CH_AC_RECORD_DESC_DEF(oob_ac, NFC_AC_CPS_ACTIVE, 1, "0", 0);
-	NFC_NDEF_CH_HS_RECORD_DESC_DEF(hs_rec, NFC_NDEF_CH_MSG_MAJOR_VER,
-				       NFC_NDEF_CH_MSG_MINOR_VER, 1);
+static void paring_key_process(void)
+{
+	int err;
 
-	ch_records.ac = &NFC_NDEF_CH_AC_RECORD_DESC(oob_ac);
-	ch_records.carrier = &NFC_NDEF_LE_OOB_RECORD_DESC(oob_rec);
-	ch_records.cnt = 1;
+	if (events[NFC_TNEP_EVENTS_NUMBER].state == K_POLL_STATE_SIGNALED) {
+		err = paring_key_generate();
+		if (err) {
+			printk("Pairing key generation error: %d\n", err);
+		}
 
-	err = nfc_ndef_ch_msg_hs_create(&NFC_NDEF_MSG(hs_msg),
-					&NFC_NDEF_CH_RECORD_DESC(hs_rec),
-					&ch_records);
-	if (err) {
-		return err;
+		k_poll_signal_reset(events[NFC_TNEP_EVENTS_NUMBER].signal);
+		events[NFC_TNEP_EVENTS_NUMBER].state = K_POLL_STATE_NOT_READY;
 	}
+}
 
-	err = nfc_ndef_msg_encode(&NFC_NDEF_MSG(hs_msg),
-				  nfc_t4t_ndef_file_msg_get(ndef_msg_buf),
-				  &ndef_size);
-	if (err) {
-		return err;
+static void bond_find(const struct bt_bond_info *info, void *user_data)
+{
+	int err;
+
+	/* Filter already connected peers. */
+	for (size_t i = 0; i < CONFIG_BT_MAX_CONN; i++) {
```

Comment:
Why the for-loop? `i` is not used.

Reply:
Good catch. This loop is no needed here what is more the conn have to be unref after calling function 'bt_conn_lookup_addr_le'. Done